### PR TITLE
Abocquet/media selector

### PR DESCRIPTION
**Pull request by abocquet**

### CODE REVIEW

File: frontend/src/components/associations/settings/AssociationLogoSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+          ) : (
+            "Aucun logo pour l'instant"
+          )}
+        </p>
+
+        <button
+          className="btn btn-primary"
+          onClick={() => setShowSelectorModal(true)}
+        >
+          Séléctionner un logo
```
Comment (Piwakk) [inline frontend/src/components/associations/settings/AssociationLogoSettings.tsx:48]:
```suggestion
          Sélectionner un logo
```

File: frontend/src/components/associations/medias/PreviewCard.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    overlayInformation = false,
+    onClick = () => {},
+    ...props
+  } /* {
+  media: Media;
+  showPreview?: boolean;
+  showDescription?: boolean;
+  showTags?: boolean;
+  onClick?: () => void;
+}*/
```
Comment (Piwakk) [inline frontend/src/components/associations/medias/PreviewCard.tsx:32]: No type finally?

File: frontend/src/components/associations/medias/PreviewCard.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+/**
+ * Generic component to present a media in a card
+ * @param media the media to display
+ * @param showPreview show the preview
+ * @param showDescription show the preview
+ * @param showTags show the preview
+ * @param overlayInformation true to show the information on the image on hover
+ * @param onClick callback in case the card is clicked
+ * @param props additional JSX props
```
Comment (Piwakk) [inline frontend/src/components/associations/medias/PreviewCard.tsx:14]:
```suggestion
 * @param props additional JSX props passed to the `Card` component
```

File: frontend/src/components/associations/medias/PreviewCard.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
+import Card from "react-bootstrap/Card";
+import { TaggableModel, TagList } from "../../utils/tags/TagList";
+import React from "react";
+import "./PreviewCard.css";
+
+/**
+ * Generic component to present a media in a card
+ * @param media the media to display
+ * @param showPreview show the preview
+ * @param showDescription show the preview
```
Comment (Piwakk) [inline frontend/src/components/associations/medias/PreviewCard.tsx:10]:
```suggestion
 * @param showDescription show the description
```

File: frontend/src/components/associations/medias/PreviewCard.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+ * @param showTags show the preview
```
Comment (Piwakk) [inline frontend/src/components/associations/medias/PreviewCard.tsx:11]:
```suggestion
 * @param showTags show the tags
```

File: frontend/src/components/associations/settings/AssociationLogoSettings.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+import { queryCache } from "react-query";
+import { Media } from "../../../models/associations/media";
+
+export const AssociationLogoSettings = ({ association }) => {
+  const { data: logo, status, error } = useBetterQuery<Media>(
+    ["association.logo.get", association.logo],
+    api.medias.get
+  );
+
+  const [showSelectModal, setShowSelectorModal] = useState(false);
```
Comment (Piwakk) [inline frontend/src/components/associations/settings/AssociationLogoSettings.tsx:14]: Not the same name

File: frontend/src/components/utils/MediaSelector.tsx [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+import { SidebarSeparator } from "./sidebar/Sidebar";
+import { SidebarDateSelector } from "./sidebar/SidebarDateSelector";
+import { Association } from "../../models/associations/association";
+
+/**
+ * A generic way to select one (maybe more in the future) media from an
+ * association media galery
+ * @param association the association from which we want the medias
+ * @param imageOnly if we want to only display  images
+ * @param setMedia a callback called when the user validate a media
```
Comment (Piwakk) [inline frontend/src/components/utils/MediaSelector.tsx:19]:
```suggestion
 * @param setMedia a callback called when the user validates a media
```

File: frontend/src/components/associations/medias/PreviewCard.tsx (hunk 2 above)
Comment (abocquet) [reply frontend/src/components/associations/medias/PreviewCard.tsx:32]: Typing ...props is a real pain

Comment (Piwakk) [reply frontend/src/components/associations/medias/PreviewCard.tsx:32]: https://github.com/typescript-cheatsheets/react-typescript-cheatsheet/blob/master/README.md#useful-react-prop-type-examples

Does that help you?

**Pull request merged**